### ISSUE TITLE
User can set storage limits for files

### ISSUE THREAD

**Issue by holmesworcester** [TryQuiet/monorepo#529]
We don't want the default behavior of Quiet to be "files stored until they fill up your storage", so there should be some limit. Typical users should not have to worry about this and the default behavior should fit many situations. However, users should have control over how much device storage their Quiet uses.

1. Quiet should timestamp locally when file downloads were initiated, from that user's point of view. (Not completed, because we don't want uncompleted files to stick around forever, and not initially created, because we don't need to trust other users.)
2. Quiet should also keep track locally of what files belong to the user's own messages, vs what files belong to other users' messages. 
3. Users should be able to optionally set separate storage limits for their own files, and others' files, for a given community. 
4. Whenever a new download is begun, and total disk space once that download is complete would exceeds max storage for that category, we should remove the oldest timstamped files in that category by "unpinning" them and running ipfs garbage collection.

Future work:

1. also allow users to set time limits for others' files, to keep storage levels well below maximum tolerable limits. 
2. add optional (active by default) protection against filling up device storage
3. add admin-configurable, per-user limits on the number of messages you can send i.e. anti-spam rules

Design proposal:

1. There should be a "Images & files" tab in settings
2. There should be a section titled "Set storage limits..."
3. There should be two checkboxes: "For my images & files" and "For others' images & files" 
4. If the checkbox is checked, a size limit selector is available, with unlimited choices starting at 100MB in 100MB intervals. 5. The default for both should be large, say 2000MB.

**Comment by holmesworcester** [comment]
One note is that OrbitDB data should not affect this limit, and we should never delete things from OrbitDB.

**Comment by jgaylor** [comment]
Here's the latest [Loom](https://www.loom.com/share/2a36d9f3654843e99b282c740e6ef6b8) and [Figma](https://www.figma.com/file/XUzVGC04b3j7Pj31DRN814/File-storage-limits?type=design&node-id=832-6001&mode=design&t=P4NbUvc7OzSmqeOY-4)

Still some things undone, but we've decided to keep some placeholders for now, as we have a solid direction for how this will work.
